File: backend/apps/usuarios/repositories/servicio_repository.py
from ..models import Servicio


class ServicioRepository:

    @staticmethod
    def listar_todos():
        return Servicio.objects.all()

    @staticmethod
    def buscar_por_id(servicio_id):
        return Servicio.objects.get(pk=servicio_id)

    @staticmethod
    def de_profesional(profesional_id):
        return Servicio.objects.filter(profesional_id=profesional_id)

File: backend/src/repositories/servicio.repository.ts
import { Servicio } from "../models/Servicio";

export class ServicioRepository {
  static listarTodos() {
    return Servicio.find().lean();
  }
  static buscarPorId(id: string) {
    return Servicio.findById(id);
  }
  static deProfesional(profesionalId: string) {
    return Servicio.find({ profesional: profesionalId }).lean();
  }
  static crear(data: any) {
    return Servicio.create(data);
  }
}
